cardinal_toe_app/src/app/components/location-menu/location-menu.component.ts

```diff
@@ -0,0 +1,54 @@
+import { Input, Component, Output, EventEmitter, OnChanges, SimpleChanges } from '@angular/core';
+import { Place } from '../../interfaces/place';
+import { DialogModule } from 'primeng/dialog';
+import { ButtonModule } from 'primeng/button';
+import { NgIf, TitleCasePipe } from '@angular/common';
+
+@Component({
+  selector: 'app-location-menu',
+  imports: [ DialogModule, ButtonModule, NgIf, TitleCasePipe ],
+  standalone: true,
+  templateUrl: './location-menu.component.html',
+  styleUrls: ['./location-menu.component.scss']
+})  
+export class LocationMenuComponent implements OnChanges {
+  @Input() display: boolean = false;
+  @Input() selectedDestination?: Place | undefined;
+  @Input() currentLocation?: boolean | undefined;
+  @Output() changeDisplay = new EventEmitter<boolean>();
+  @Output() changeNode = new EventEmitter<Place>();
+
+  localDestination: Place | undefined;
```

Comment:
generally prefer use of `null` over `undefined` - they evaluate to the same.

Reply:
`undefined` used as the library Primeng needs input with undefined. Will look into this.